BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m

```diff
@@ -559,7 +560,10 @@ - (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didRece
             [alertController addAction:okAction];
             [self presentViewController:alertController animated:YES completion:nil];
         }
-        [self.activityIndicator stopAnimating];
+        BOOL isMainThread = [NSThread isMainThread];
```

Comment:
This *has* to happen on the main thread, so we don't have to know if that's where we're coming from.

Won’t this also crash if the alerts above or below this aren't on the main thread? This will only save us from a crash if we have no alerts to present. It looks like there's an assumption this method gets called on the main thread, but that's false.